client/src/rpb/granular_view.js

```diff
@@ -153,112 +68,96 @@ class GranularView extends React.Component {
-                />
-              </p>
-            )}
-            <ul className="pagination">
-              {_.map(pages, (data, ix) => (
-                <li
-                  key={ix}
-                  className={classNames(ix === page_num && "active")}
-                >
-                  <span
-                    tabIndex={0}
-                    style={
-                      ix === page_num
-                        ? {
-                            color:
-                              window.infobase_color_constants.textLightColor,
-                          }
-                        : null
-                    }
-                    disabled={page_num === ix}
-                    role="button"
-                    onClick={
-                      ix === page_num
-                        ? null
-                        : () => {
-                            on_set_page(ix);
-                            this.refs.table.focus();
-                          }
-                    }
-                    onKeyDown={
-                      ix === page_num
-                        ? null
-                        : (e) => {
-                            if (e.keyCode === 13 || e.keyCode === 32) {
-                              on_set_page(ix);
-                              this.refs.table.focus();
+      <DisplayTable
+        data={table_data}
+        column_configs={column_configs}
+        util_components={{
+          rpb_group_data: (
+            <DropdownMenu
+              opened_button_class_name={"btn-ib-opened"}
+              closed_button_class_name={"btn-ib-light"}
+              key={"rpb_group_data"}
+              button_description={text_maker("group_data")}
+              custom_dropdown_trigger={`${text_maker("group_by")}: ${filter}`}
+              dropdown_content={
+                <div className="group_filter_dropdown">
+                  {_.map(group_filter_options, (group) => (
+                    <div key={group.id}>
+                      <span key={group.id} style={{ fontWeight: 700 }}>
+                        {group.display}
+                      </span>
+                      {_.map(group.children, (child) => (
+                        <div key={`${group.id}_${child.id}`}>
+                          <input
+                            type={"radio"}
+                            value={child.id}
+                            id={child.id}
+                            name={"rpb_group_filter"}
+                            key={child.id}
+                            onClick={(evt) => {
+                              const [
+                                dim_key,
+                                filt_key,
+                              ] = evt.target.value.split("__");
+                              on_set_filter({
+                                filter: filt_key,
+                                dimension: dim_key,
+                              });
+                            }}
```

Comment:
Don't need to access the value off the event object, it's directly available in scope as `child.id`. Code's a bit more clear as
```
onClick={() => {
  const [
    dim_key,
    filt_key,
  ] = child.id.split("__");
  on_set_filter({
    filter: filt_key,
    dimension: dim_key,
  });
}}
```

That said, storing two values in `id` and then doing all these splits/comparisons using `__` is a bit hacky. I know that's how the RPB was originally written, but could you update the `state_and_data.js` and this code so that `dim_key` and `filter_key` are available individually as properties of `child`?